File: backend/components/metiers/administration_publique/services/core/samples/sample_service.js
// sample_service.js
// Exemple de service (sample) pour le module core services threed.
// Sert de référence pour l'intégration, la documentation et les tests.

class SampleService {
  constructor(options = {}) {
    this.options = options;
    this.state = 'ready';
  }

  /**
   * Initialise le sample service avec des paramètres de test.
   * @param {Object} config - Configuration de test.
   */
  init(config) {
    this.config = config;
    this.state = 'initialized';
    return true;
  }

  /**
   * Exécute la fonctionnalité principale du sample service.
   * @param {any} data - Données à traiter.
   * @returns {any} Résultat du traitement sample.
   */
  run(data) {
    return { processed: true, data, config: this.config, state: this.state };
  }
}

module.exports = SampleService;

File: backend/components/metiers/threed/services/core/samples/sample_service.py
"""
sample_service.py
Exemple de service (sample) pour le module core services threed (Python).
Sert de référence pour l'intégration, la documentation et les tests.
"""

class SampleService:
    def __init__(self, options=None):
        self.options = options or {}
        self.state = 'ready'
        self.config = None

    def init(self, config):
        self.config = config
        self.state = 'initialized'
        return True

    def run(self, data):
        return {"processed": True, "data": data, "config": self.config, "state": self.state}
